File: CS/CustomTableOfContents/ProductsReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace CustomTableOfContents {
    public partial class ProductsReport : DevExpress.XtraReports.UI.XtraReport {
        public ProductsReport() {
            InitializeComponent();
        }

        private void tcCategoryName_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            XRControl label = sender as XRControl;
            label.Tag = String.Format("Target_{0}", GetCurrentColumnValue("CategoryID"));
        }

    }
}

File: VB/CustomTableOfContents/ProductsReport.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace CustomTableOfContents

    Public Partial Class ProductsReport
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub tcCategoryName_BeforePrint(ByVal sender As Object, ByVal e As CancelEventArgs)
            Dim label As XRControl = TryCast(sender, XRControl)
            label.Tag = String.Format("Target_{0}", GetCurrentColumnValue("CategoryID"))
        End Sub
    End Class
End Namespace
